### PR TITLE
DOCS-#4628: add to_parquet partial support notes

### LINKED ISSUE #4628: DOCS: Document partial support for to_pandas.
Currently, the [supported APIs page](https://modin.readthedocs.io/en/stable/supported_apis/dataframe_supported.html) says that Modin always defaults to pandas for `to_parquet`. Per #4624, we should document that Modin on Ray has a parallel implementation of `to_parquet` for the cases not excluded in [this check](https://github.com/modin-project/modin/blob/4ec7f6347903f9133c65ebc5b6e0e15553b98577/modin/core/execution/ray/implementations/pandas_on_ray/io/io.py#L241).

The file to edit is [here](https://github.com/modin-project/modin/blob/7b83e3b26674ff764c412864f886fcb34206bc29/docs/supported_apis/dataframe_supported.rst).

### PR DESCRIPTION
**Pull request by d33bs**
Documentation update to specify Dask vs Ray differences and parallel implementation cases of to_parquet with Ray as per discussion within #4624.

Note: no tests added as update only changes documentation.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4628 
- [ ] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: docs/supported_apis/dataframe_supported.rst [hunk 1]
```diff
@@ -418,7 +418,14 @@ default to pandas.
...
+| ``to_parquet``             | `to_parquet`_             | P                      | **Dask**: Defaults to Pandas implementation as a   |
+|                            |                           |                        | single file. **Ray**: Parallel implementation with |
+|                            |                           |                        | potentially multiple file output only if path      |
+|                            |                           |                        | parameter is a string, does not end with ".gz",    |
+|                            |                           |                        | ".bz2", ".zip", or ".xz" and compression parameter |
+|                            |                           |                        | is not none or "snappy". In these cases, paths     |
+|                            |                           |                        | with Modin on Ray for ``to_parquet`` are           |
+|                            |                           |                        | `directory` and not file paths.                    |
```
Comment (mvashishtha) [inline docs/supported_apis/dataframe_supported.rst:428]:
```suggestion
| ``to_parquet``             | `to_parquet`_             | P                      | **Dask**: Defaults to Pandas implementation and 
|                            |                           |                        | writes a  single output file. 
|                            |                           |                        | **Ray**: Parallel implementation only if path  |
|                            |                           |                        | parameter is a string; does not end with ".gz",    |
|                            |                           |                        | ".bz2", ".zip", or ".xz"; and compression parameter |
|                            |                           |                        | is not ``None`` or "snappy". In these cases, the |
|                            |                           |                        | ``path`` parameter specifies a directory where |
|                            |                           |                        | one file is written per row partition of the Modin dataframe. |
```

I think this wording is a little clearer. I have probably messed up the table formatting a little, or at least made some of the lines too long, so if you accept this suggestion, you might have to clean that up.

**Review by mvashishtha — COMMENTED** [review]
@d33bs thank you so much for this PR! I have left one minor comment.

**Comment by d33bs** [comment]
> @d33bs thank you so much for this PR! I have left one minor comment.

@mvashishtha Thank you, I just accepted the changes (much clearer!) and reformatted the table.

**Review by mvashishtha — APPROVED** [review]
LGTM! Thank you again for the PR.

**Review by RehanSD — APPROVED** [review]
Hi @d33bs ! Thank you so much for this PR - it looks great! Approving and merging!

**Comment by d33bs** [comment]
Thank you @mvashishtha and @RehanSD !

**Pull request merged**